packages/cli/src/lib/decorators/preconditions/node.spec.ts

```diff
@@ -0,0 +1,164 @@
+jest.mock('../../utils/process');
+
+import type Command from '@oclif/command';
+import * as dedent from 'dedent';
+import {constants} from 'os';
+import {mocked} from 'ts-jest/utils';
+import {spawnProcessOutput} from '../../utils/process';
+
+import {IsNodeVersionAbove} from './node';
+
+describe('IsNodeVersionAbove', () => {
+  const mockedSpawnProcessOutput = mocked(spawnProcessOutput);
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('when the requiredVersion is not a semver valid string', () => {
+    it('should return false and warn', async () => {
+      const fakeCommand = {
+        warn: jest.fn(),
+      };
+
+      await expect(
+        IsNodeVersionAbove('foo')((fakeCommand as unknown) as Command)
+      ).resolves.toBe(false);
+      expect(fakeCommand.warn).toHaveBeenCalledTimes(1);
+      expect(fakeCommand.warn).toHaveBeenCalledWith(dedent`
+        Required version invalid: "foo".
+        Please report this error to Coveo: https://github.com/coveo/cli/issues/new
+      `);
+    });
+  });
+
+  describe('when Node.js is not installed', () => {
+    beforeEach(() => {
+      mockedSpawnProcessOutput.mockResolvedValue({
+        exitCode: constants.errno.ENOENT,
+        stderr: '',
+        stdout: '',
+      });
+    });
+
+    it('should return false and warn', async () => {
+      const fakeCommand = {
+        id: 'foo',
+        warn: jest.fn(),
+      };
+
+      await expect(
+        IsNodeVersionAbove('0.0.1')((fakeCommand as unknown) as Command)
```

Comment:
I prolly could do a bit cleaner than the double cast with some Jest-pirouette, but I didn't think it was worth the hassle.